src/extlibs/hrvo/visualizer/hrvo_visualizer.py

```diff
@@ -0,0 +1,181 @@
+                    )
+                    relative_speed = pow(
+                        pow(relative_vel_x, 2) + pow(relative_vel_y, 2), 0.5
+                    )
+                    print(
+                        f"robot {robot_id} and robot {other_robot_id} have collided with a relative velocity of {relative_speed}m/s"
+                    )
+            else:
+                robot_list[robot_id].set_facecolor(default_robot_color)
+
+        # Update lines tracking the robot movement
+        robots_grouped = robot_pos_df.groupby("robot_id")
+        for robot_id, line in enumerate(line_list):
+            robot_df = robots_grouped.get_group(robot_id)
+            line.set_data(
+                robot_df[robot_df["frame"] <= frame]["x"],
+                robot_df[robot_df["frame"] <= frame]["y"],
+            )
+
+        # Update displayed time
+        curr_time = float(curr_frame_df["time"].head(1))
+        time_text.set_text(f"Time: {curr_time:.2f}sec")
+
+        return robot_list + line_list + [time_text] + [robot_id_text]
+
+    def toggle_pause(event):
+        nonlocal is_paused
+        if is_paused:
+            robot_anim.resume()
+        else:
+            robot_anim.pause()
+        is_paused = not is_paused
+
+    def on_plot_hover(event):
+        # Iterating over each data member plotted
+        for robot_id, line in enumerate(line_list):
+            # Searching which data member corresponds to current mouse position
+            if line.contains(event)[0]:
+                line.set_alpha(1.0)
+                robot_id_text.set_text(f"Robot {robot_id}")
+            else:
+                line.set_alpha(default_line_alpha)
+
+    # set up plot
+    fig = plt.figure()
+    fig.suptitle(file_name)
+    ax = fig.add_subplot(
+        111,
+        autoscale_on=False,
+        xlim=(min_robot_x_pos - x_offset, max_robot_x_pos + x_offset),
+        ylim=(min_robot_y_pos - y_offset, max_robot_y_pos + y_offset),
+    )
+    ax.set_aspect("equal")
+    ax.grid()
+    (line,) = ax.plot([], [], "--r")
+    # plt.subplots_adjust(left=0.3)
+    time_text = ax.text(0.5, 0.2, "", transform=plt.gcf().transFigure)
+    robot_id_text = ax.text(0.5, 0.1, "", transform=plt.gcf().transFigure)
+
+    # set-up robot circles
+    initial_frame = robot_pos_df[robot_pos_df["frame"] == 0]
+    robot_list = []
+    line_list = []
+    for robot_id in range(num_robots):
+        radius = float(
+            initial_frame[initial_frame["robot_id"] == robot_id]["radius"].head(1)
+        )
+        robot = Circle((0, 0), radius, facecolor="aqua", edgecolor="black")
+        robot_list.append(robot)
+        line_obj = ax.plot([], [], linestyle="--", alpha=default_line_alpha)[0]
+        line_list.append(line_obj)
+
+    # Start/Stop on click
+    fig.canvas.mpl_connect("button_press_event", toggle_pause)
+
+    # TODO: Allow highlighting after animation is over. Can achieve this by not passing frames to FunAnimation
```

Comment:
nit: this TODO doesn't have an issue attached, same with other TODOs in this PR

Reply:
Removed extra TODOs, and added an issue to one.